cloudant-client/src/main/java/com/cloudant/client/internal/views/RowImpl.java

```diff
@@ -58,11 +58,19 @@ public V getValue() {
 
     @Override
     public Document getDocument() {
-        return gson.fromJson(row.get("doc"), Document.class);
+        Document doc = null;
+        if(row.has("doc") && !row.has("error")) {
+            doc = gson.fromJson(row.get("doc"), Document.class);
+        }
+        return doc;
     }
 
     public <D> D getDocumentAsType(Class<D> docType) {
-        return gson.fromJson(row.get("doc"), docType);
+        D doc = null;
+        if(row.has("doc") && !row.has("error")) {
+            doc = gson.fromJson(row.get("doc"), docType);
+        }
+        return doc;
```

Comment:
If we're returning `null` here we'll need `com.cloudant.client.internal.views.ViewResponseImpl#getDocsAs` to perform a `null` check before adding to the list too.

Reply:
Fixed in 7f24237.